frontend/src/components/layouts/AppLayout.tsx

```diff
@@ -18,14 +22,29 @@ export default function AppLayout() {
 
   return (
     <>
-      <div className="font-sans min-h-screen w-full flex flex-col">
-        <SidebarProvider>
-          <AppSidebar />
-          <SidebarInset>
-            <div className="flex flex-1 flex-col gap-4 p-4">
-              <Outlet />
-            </div>
-          </SidebarInset>
+      <div
+        className={cn(
+          "font-sans min-h-screen w-full flex flex-col",
+          showBanner
+            ? "[--header-height:calc(theme(spacing.9))]" // Banner height is 36px when visible (sidebar hidden on <md width)
+            : "[--header-height:0]"
+        )}
+      >
+        <SidebarProvider className="flex flex-col">
+          {showBanner && <UpdateBanner onDismiss={dismissBanner} />}
+          <div className="flex flex-1">
+            <AppSidebar />
+            <SidebarInset>
+              <div
+                className={cn(
+                  "flex flex-1 flex-col gap-4 p-4",
+                  showBanner && "mt-14 md:mt-9" // Banner height is 36px with 1 line (>=md width) and 56px with 2 lines (<md width)
```

Comment:
Do we really need 2-line banners?

Reply:
I think it's good to have that, otherwise we can't convey any info in 3-4 words:

<img width="323" alt="Screenshot 2025-05-08 at 6 21 27 PM" src="https://github.com/user-attachments/assets/f622d33f-7bb4-42dc-ad15-5b90446dccd9" />